src/main/kotlin/domain/lotto/domain/Lotto.kt

```diff
@@ -0,0 +1,37 @@
+package domain.lotto.domain
+
+import domain.lotto.error.InvalidLottoNumberSizeException
+import domain.lotto.strategy.LottoShuffleStrategy
+import global.strategy.split.SplitStrategy
+
+@JvmInline
+value class Lotto private constructor(private val _lotto: Set<LottoNumber>) {
+    val lotto: Set<LottoNumber>
+        get() = _lotto.toSortedSet()
+
+    fun match(other: Lotto): Int = other.lotto.count { lotto.contains(it) }
+
+    companion object {
+        const val PRICE = 1_000
+        private const val FROM_INDEX = 0
+        private const val TO_INDEX = 6
+
+        fun of(shuffleStrategy: LottoShuffleStrategy): Lotto {
+            val lottoNumbers = LottoNumber.values()
+            val shuffledLottoNumbers = shuffleStrategy.shuffle(lottoNumbers)
+            val selectedLottoNumbers = shuffledLottoNumbers.subList(FROM_INDEX, TO_INDEX)
+            return of(selectedLottoNumbers.toSortedSet())
+        }
+
+        fun of(lotto: String, splitStrategy: SplitStrategy): Lotto =
+            of(
+                splitStrategy.split(lotto)
+                    .map { LottoNumber.of(it.toInt()) }
+                    .toSortedSet()
+            )
+
+        fun of(lotto: Set<LottoNumber>): Lotto =
+            if (lotto.size == TO_INDEX) Lotto(lotto.toSortedSet())
```

Comment:
```suggestion
            if (lotto.size < TO_INDEX || lotto.size > TO_INDEX)  {
                throw InvalidLottoNumberSizeException(lotto.size)
            }
            
            Lotto(lotto.toSortedSet())
```

와 같이 표현하면 else 를 사용안할 수 있지 않을까요?

Reply:
제가 문법을 잘 이해했는지 헷갈리는데, 식으로 표현하는 else 를 사용하게 되는 것 같습니다.